src/store.js

```diff
@@ -158,10 +158,16 @@ export class Store {
       assert(path.length > 0, 'cannot register the root module by using registerModule.')
     }
 
-    this._modules.register(path, rawModule)
-    installModule(this, this.state, path, this._modules.get(path))
-    // reset store to update getters...
-    resetStoreVM(this, this.state)
+    const moduleResolved = module => {
+      this._modules.register(path, module.default || module)
+      installModule(this, this.state, path, this._modules.get(path))
+        // reset store to update getters...
+      resetStoreVM(this, this.state)
+      return module
+    }
+
+    if (isPromise(rawModule)) return rawModule.then(moduleResolved)
+    return moduleResolved(rawModule)
```

Comment:
[Vuex requires Promise to work](https://github.com/vuejs/vuex/blob/dev/src/store.js#L12). So we can consider that Promise is always there 🙂